dunereco/FDSensOpt/NeutrinoAngularRecoAlg/NeutrinoAngularRecoAlg.cc

```diff
@@ -174,6 +177,200 @@ dune::AngularRecoOutput NeutrinoAngularRecoAlg::CalculateNeutrinoAngle(const art
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+dune::AngularRecoOutput NeutrinoAngularRecoAlg::CalculateNeutrinoAngle(const art::Event &event, const Point_t &vertex){
+    std::vector<PolarFitOutput> output_vec;
+    //Getting all the reconstructed 2D hits
+    const std::vector<art::Ptr<recob::Hit>> hits(dune_ana::DUNEAnaEventUtils::GetHits(event, fHitLabel));
+    //Sorting all the hits per view
+    //Taking as reference the views for positive drift -> U- == V+ and V- == U+
+    std::map<geo::View_t, std::vector<art::Ptr<recob::Hit>>> hits_per_view;
+    for(art::Ptr<recob::Hit> const& hit: hits){
+        geo::View_t view = GetTargetView(hit);
+
+        if(hits_per_view.count(view) == 0){
+            hits_per_view[view] = {};
+        }
+        hits_per_view[view].push_back(hit);
+    }
+
+    for(auto const& [view, hits] : hits_per_view){
+        PolarFitOutput output = FitViewHits(event, view, hits, vertex);
+        output_vec.push_back(output);
+    }
+
+    const std::vector<std::pair<double, double>> start_values = { //Changng the starting point seems to help convergence in some cases
+        {0.5, 0.5},
+        {0.5, 2},
+        {2, 0.5},
+        {2, 2},
+    };
```

Comment:
It would be good to make these static global variables (at the top of the source file) or configurable through fcl